Fixed this app

iaronaraujo opened the pull request:
#### What is the purpose of this pull request?

Refactored the whole app and made it work. It is now possible to view product reviews on the product page.

#### How should this be manually tested?

[Workspace](https://iaronaraujo--motorolaus.myvtex.com/smartphones-moto-g6-play/p)

#### Types of changes

* [x] Bug fix (a non-breaking change which fixes an issue)
* [ ] New feature (a non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Requires change to documentation, which has been updated accordingly.

klzns commented on node/package.json:
```diff
@@ -4,10 +4,12 @@
   "main": "index.js",
   "dependencies": {
+    "apollo-server": "^2.8.1",
+    "apollo-server-errors": "^2.3.1",
```
Remove this dependency with `yarn remove apollo-server-errors`
```suggestion
    "apollo-server-errors": "^2.3.1",
```

klzns commented on react/components/withGetConfig.tsx:
```diff
@@ -3,11 +3,13 @@ import { graphql } from 'react-apollo'
-})
+
+const withGetConfig = graphql<{}, Settings>(getConfig)
+//const withGetConfig = graphql<{}, Settings>(getConfig, {options: {ssr:false}})
```
```suggestion
```

iaronaraujo commented:
More details about what was fixed:
- Reviews now appear correctly!
- If the bazaarvoice query results in error, its component won't be displayed
- Refactored the whole code
- `next review` button is now working properly and showing the correct numbers
- Fixed the time of the reviews
- Fixed the number displayed next to the stars
- Fixed histogram

jgfidelis commented on react/components/Stars.tsx:
```diff
@@ -0,0 +1,85 @@
+      <div className="review__rating--inactive nowrap">
+        {[0, 1, 2, 3, 4].map((_, i) => {
+          return i <= 3 ? (
+            <svg
```
you could break this SVG into smaller components because you render the same stuff multiple times.

You could then do <Star fill="#eee" height="20" width="20" />, something like this

jgfidelis commented on react/components/Stars.tsx:
```diff
@@ -0,0 +1,85 @@
+
+const Stars: FunctionComponent<StarsProps> = ({ rating }) => {
+  return (
+    <div className="review__rating--stars dib relative v-mid mr2">
```
no css modules?

iaronaraujo commented:
Created a task related to this app's documentation https://app.clubhouse.io/vtex/story/18083/update-docs-of-bazzarvoice

jgfidelis commented on react/components/Stars.tsx:
```diff
@@ -0,0 +1,34 @@
+
+const Stars: FunctionComponent<StarsProps> = ({ rating }) => {
+  return (
+    <div className={`${styles.review__rating_stars} dib relative v-mid mr2`}>
```
I don't know, this may be a breaking change, was the class `review__rating_stars` already inserted before?

@klzns qual sua opinião?

jgfidelis commented on react/components/Star.tsx:
```diff
@@ -1,26 +1,32 @@
         transform="translate(0)"
       />
-    </svg>
+    </svg> // se o review.metrics.rating for 4, preenche 4 estrelas
```
remove comment

jgfidelis commented on react/components/Star.tsx:
```diff
@@ -1,26 +1,32 @@
-import styles from '../styles.css'
 
-const Star: FunctionComponent<StarProps> = ({ fill }) => {
+const Star: FunctionComponent<StarProps> = ({key, fill, className, size}) => {
```
key prop should not be passed here

jgfidelis commented on react/components/Star.tsx:
```diff
@@ -1,26 +1,32 @@
     <svg
-      className={`${styles.star} ${fill ? styles['star--filled'] : ''}`}
+      className={className}
+      key={key}
```
remove
